src/Backends/SystemUpdate.vala

```diff
@@ -0,0 +1,255 @@
+
+        check_for_updates.begin (false, true);
+
+        Timeout.add_seconds ((uint) settings.get_int64 ("refresh-interval"), () => {
+            check_for_updates.begin (false, true);
+            return Source.CONTINUE;
+        });
+    }
+
+    public async void check_for_updates (bool force, bool notify) throws DBusError, IOError {
+        if (current_state.state != UP_TO_DATE && current_state.state != AVAILABLE && !force) {
+            return;
+        }
+
+        update_state (CHECKING);
+
+        try {
+            yield task.refresh_cache_async (force, null, progress_callback);
+        } catch (Error e) {
+            warning ("Failed to refresh cache: %s", e.message);
+        }
+
+        try {
+            available_updates = (yield task.get_updates_async (Pk.Filter.NONE, null, progress_callback)).get_package_sack ();
+
+            settings.set_int64 ("last-refresh-time", new DateTime.now_utc ().to_unix ());
+
+            if (available_updates == null || available_updates.get_size () == 0) {
+                update_state (UP_TO_DATE);
+                return;
+            }
+
+            string[] package_names = {};
+
+            foreach (var package in available_updates.get_array ()) {
+                package_names += package.get_name ();
+            }
+
+            update_details = {
+                package_names,
+                0 //FIXME: Is there a way to get update size from PackageKit
+            };
+
+            if (notify) {
+                var notification = new Notification (_("Update Available"));
+                notification.set_body (_("A system update is available"));
+                notification.set_icon (new ThemedIcon ("software-update-available"));
+                notification.set_default_action (Application.ACTION_PREFIX + Application.SHOW_UPDATES_ACTION);
+
+                GLib.Application.get_default ().send_notification (null, notification);
+            }
+
+            update_state (AVAILABLE);
+        } catch (Error e) {
+            warning ("Failed to get available updates: %s", e.message);
+            update_state (UP_TO_DATE);
+        }
+    }
+
+    public async void update () throws DBusError, IOError {
+        if (current_state.state != AVAILABLE) {
+            return;
+        }
+
+        cancellable.reset ();
+
+        update_state (DOWNLOADING);
+
+        try {
+            yield task.update_packages_async (available_updates.get_ids (), cancellable, progress_callback);
+
+            Pk.offline_trigger (REBOOT);
+
+            var notification = new Notification (_("Restart Required"));
+            notification.set_body (_("Please restart your system to finalize updates"));
+            notification.set_icon (new ThemedIcon ("system-reboot"));
+            notification.set_default_action (Application.ACTION_PREFIX + Application.SHOW_UPDATES_ACTION);
+
+            GLib.Application.get_default ().send_notification (null, notification);
```

Comment:
Same here for setting an ID. All of these updates notifications should probably replace each other since they're mutually exclusive states
```suggestion
            GLib.Application.get_default ().send_notification ("system-update", notification);
```